process/block/shardblock.go

```diff
@@ -742,17 +746,55 @@ func (sp *shardProcessor) restoreMetaBlockIntoPool(mapMiniBlockHashes map[string
 
 	for metaBlockHash, miniBlockHashes := range mapMetaHashMiniBlockHashes {
 		for _, miniBlockHash := range miniBlockHashes {
-			sp.processedMiniBlocks.AddMiniBlockHash(metaBlockHash, string(miniBlockHash))
+			//TODO: Check if needed, how to set the real index (metaBlock -> ShardInfo -> ShardMiniBlockHeaders -> TxCount)
+			indexOfLastTxProcessed := common.MaxIndexOfTxInMiniBlock
```

Comment:
we should add here the correct miniBlock txhash index.
also please extract this into a method

Reply:
Done